### PR TITLE
:sparkles: Add controlPlaneAvailabilityZones and controlPlaneOmitAvailabilityZone variables

### LINKED ISSUE #154: Implement ControlPlaneAvailabilityZones variable for the OpenStack provider
/kind feature

As a user of Cluster Stacks, I want to choose in which AZ my control-plane nodes will be spawned. In the CAPO, it is possible to specify `OpenStackCluster.spec.controlPlaneAvailabilityZones` as a list of AZs. There is also a `controlPlaneOmitAvailabilityZone` boolean field which can be also exposed.

### PR DESCRIPTION
**Pull request by chess-knight**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #154

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: providers/openstack/scs/cluster-class/templates/cluster-class.yaml [hunk 1]
```diff
@@ -588,6 +604,34 @@ cre ate group names like oidc:engineering and oidc:infra."
             path: "/spec/template/spec/networkMTU"
             valueFrom:
               variable: network_mtu
+    - name: controlPlaneAvailabilityZones
+      description: "Sets the availability zones which control plane machines may be deployed to."
+      enabledIf: {{ `"{{ if .controlPlaneAvailabilityZones }}true{{end}}"` }}
```
Comment (gtema) [inline providers/openstack/scs/cluster-class/templates/cluster-class.yaml:609]: I hate helm style: double template-ing, double if-ing, and then you get a patch that you apply in your head in order to get what is going to happen. This is such a broken concept.

**Review by gtema — APPROVED** [review]
I am barely able to review this: templates over "if"s over templates with "if"s generating patch and no tests for this. It looks fine, but would be great to have possibility to see it in tests

File: providers/openstack/scs/cluster-class/templates/cluster-class.yaml (hunk 1 above)
Comment (chess-knight) [reply providers/openstack/scs/cluster-class/templates/cluster-class.yaml:609]: Yeah, it looks strange, but we have to do it like this for now. The first templating( ``{{`...`}}`` )is for the helm to ignore the second one. Helm is a tool which we use here for the cluster-stacks packaging. And the second one comes directly from CAPI ClusterClass, see https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/write-clusterclass#optional-patches.

Comment (gtema) [reply providers/openstack/scs/cluster-class/templates/cluster-class.yaml:609]: I can interpolate this and get what goes where, but it sucks too much energy and is very error-prone without real functional (or at least unit) test

**Pull request merged**